include/ginkgo/core/distributed/neighborhood_communicator.hpp

```diff
@@ -0,0 +1,162 @@
+// SPDX-FileCopyrightText: 2017 - 2024 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#ifndef GKO_PUBLIC_CORE_DISTRIBUTED_NEIGHBORHOOD_COMMUNICATOR_HPP_
+#define GKO_PUBLIC_CORE_DISTRIBUTED_NEIGHBORHOOD_COMMUNICATOR_HPP_
+
+
+#include <ginkgo/config.hpp>
+
+
+#if GINKGO_BUILD_MPI && !GINKGO_HAVE_OPENMPI_PRE_4_1_X
```

Comment:
I think this will create problems in applications: Using a single source code for <4.1.x and >4.1.x will not be possible, as the neighborhood communicator symbols are just not in the library anymore for <4.1.x. I think it might make sense to throw `GKO_NOT_SUPPORTED` inside the member functions for OpenMPI versions < 4.1.x instead.

Reply:
Yes, you're right that is the better approach. Although I don't think it would create an issue for applications.